node/pkg/reporter/reporter.go

```diff
@@ -74,42 +66,31 @@ func (r *Reporter) Run(ctx context.Context) {
 }
 
 func (r *Reporter) report(ctx context.Context) error {
-	pairs := ""
-	for _, submissionPair := range r.SubmissionPairs {
-		pairs += submissionPair.Name + ","
-	}
-	submissionData, err := request.Request[[]SubmissionData](request.WithEndpoint(r.DalEndpoint+"/latest-data-feeds/"+pairs), request.WithTimeout(10*time.Second), request.WithHeaders(map[string]string{"X-API-Key": r.DalApiKey, "Content-Type": "application/json"}))
-
-	if err != nil {
-		return err
-	}
-
 	var feedHashes [][32]byte
 	var values []*big.Int
 	var timestamps []*big.Int
 	var proofs [][]byte
 
-	for _, data := range submissionData {
-		intValue, valueErr := strconv.ParseInt(data.Value, 10, 64)
-		if valueErr != nil {
-			log.Error().Str("Player", "Reporter").Err(valueErr).Msgf("failed to parse value in data: %v", data.Symbol)
-			continue
-		}
+	for _, pair := range r.SubmissionPairs {
+		if value, ok := r.LatestData.Load(pair.Name); ok {
+			submissionData, err := processDalWsRawData(value)
+			if err != nil {
+				log.Error().Str("Player", "Reporter").Err(err).Msg("failed to process dal ws raw data")
+				continue
+			}
 
-		timestampValue, timestampErr := strconv.ParseInt(data.AggregateTime, 10, 64)
-		if timestampErr != nil {
-			log.Error().Str("Player", "Reporter").Err(timestampErr).Msgf("failed to parse timestamp in data: %v", data)
-			continue
+			feedHashes = append(feedHashes, submissionData.feedHash)
+			values = append(values, big.NewInt(submissionData.value))
+			timestamps = append(timestamps, big.NewInt(submissionData.aggregateTime))
+			proofs = append(proofs, submissionData.proof)
+		} else {
+			log.Error().Str("Player", "Reporter").Msgf("latest data for pair %s not found", pair.Name)
```

Comment:
I think we can make logic inside iteration concurrent since LatestData is sync map. how do you say  😁

Reply:
done!